### PR TITLE
Component camera improvements

### PR DESCRIPTION
**Pull request by mariofv**
What's new:
- Camera focus (`F` key) works again. It focus in the selected game object in hierarchy.
- ComponentCamera updates its frustum position using its owner translation.

### CODE REVIEW

File: Engine/Component/ComponentTransform.h [hunk 1]
```diff
@@ -39,6 +41,8 @@ class ComponentTransform : public Component
 
 	float4x4 model_matrix = float4x4::identity;
 	float4x4 global_model_matrix = float4x4::identity;
+
+	friend class ComponentCamera;
```
Comment (yalania) [inline Engine/Component/ComponentTransform.h]: Why using friend?

Comment (mariofv) [reply Engine/Component/ComponentTransform.h]: We want to bind in `ComponentCamera` frustum position (lets call it **FP**) and the owner transform position (lets call it **TP**). Considering that there are a lot of methods in `ComponentCamera` that modify FP (`Orbit`, `MoveXXX`, ...), I approached the problem in this way:
1. In `ComponentCamera` the only position updated is TP.
2. In every frame, I set FP with TP values.
3. I generate corresponding camera matrices.

With this approach I ensure that TP == FP and also that changes to TP outside `ComponentCamera` are also reflected in FP. But this approach has the problem of accessing the private member `translation` of `ComponentTransform`.

So, to solve that I befriended `ComponentCamera` with `ComponentTransform` class. Another possible solution could be using Getters/Setters in `ComponentCamera`.

Comment (yalania) [reply Engine/Component/ComponentTransform.h]: You created GetTranslation, can't you use it?

Comment (mariofv) [reply Engine/Component/ComponentTransform.h]: I guess :D I'm on it!

**Review by yalania — APPROVED** [review]

**Pull request merged**